sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathAccessControlEntry.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace Azure.Storage.Files.DataLake.Models
+{
+    /// <summary>
+    /// Represents an access control in a file access control list.
+    /// </summary>
+    public class PathAccessControlEntry
+    {
+        /// <summary>
+        /// Indicates whether this is the default entry for the ACL.
+        /// </summary>
+        public bool DefaultScope { get; set; }
+
+        /// <summary>
+        /// Specifies which role this entry targets.
+        /// </summary>
+        public AccessControlType AccessControlType { get; set; }
+
+        /// <summary>
+        /// Specifies the entity for which this entry applies.
+        /// Must be omitted for types mask or other.  It must also be omitted when the user or group is the owner.
+        /// </summary>
+        public string EntityId { get; set; }
+
+        /// <summary>
+        /// Specifies the permissions granted to this entry.
+        /// </summary>
+        public RolePermissions Permissions { get; set; }
+
+        /// <summary>
+        /// Empty constructor.
+        /// </summary>
+        public PathAccessControlEntry() { }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="accessControlType">Specifies which role this entry targets.</param>
+        /// <param name="permissions">Specifies the permissions granted to this entry.</param>
+        /// <param name="defaultScope">Indicates whether this is the default entry for the ACL.</param>
+        /// <param name="entityId">Specifies the entity for which this entry applies.</param>
+        public PathAccessControlEntry(
+            AccessControlType accessControlType,
+            RolePermissions permissions,
+            bool defaultScope = false,
+            string entityId = null)
+        {
+            DefaultScope = defaultScope;
+            AccessControlType = accessControlType;
+            EntityId = entityId;
```

Comment:
add check for the entityId != null, if true then the accessControlType has to be user or group.

Reply:
fixed.